### ISSUE TITLE
clBuildProgram failed

### ISSUE THREAD

**Issue by MOOnLyer** [s-u/OpenCL#9]
My machine:
OS: Mac OS X 10.15

R info:
R version: 4.1.2
Package version: GitHub OpenCL_0.2-3

I am trying to follow the toy example posted under oclRun. However, when I proceed to the step "k.dnorm <- oclSimpleKernel(ctx, "dnorm", code)", an error pops out stating "Error in oclSimpleKernel(ctx, "dnorm", code) : clBuildProgram failed (with -11) "

I am wondering how I could solve it.

Many thanks!

**Comment by aaronpuchert** [comment]
That should be `CL_BUILD_PROGRAM_FAILURE`. Does it not print a build log before that? It would be strange if `clGetProgramBuildInfo` didn't provide us with an error message in this case.

**Comment by aaronpuchert** [comment]
For example when I remove the trailing brace from the `oclRun` example:
```
> k.dnorm <- oclSimpleKernel(ctx, "dnorm", code)
input.cl:12:75: error: expected '}'
input.cl:8:6: note: to match this '{'

Error in oclSimpleKernel(ctx, "dnorm", code) : 
  clBuildProgram failed (with -11)
```
which tells us what the problem is:
```
input.cl:12:75: error: expected '}'
input.cl:8:6: note: to match this '{'
```

**Comment by MOOnLyer** [comment]
> 

Well, I do not know which part went wrong. But, I tried to remove the trailing brace, and it does not provide any error message either.

**Comment by aaronpuchert** [comment]
With abdd6751aa973783bb5d4587b2dcfc35c5e0e491 we have more diagnostics, could you build that (e.g. master) and try again? Perhaps fetching the build log fails, we have so far just ignored that.

Other than that, have you tried running OpenCL workloads outside of R?

**Comment by MOOnLyer** [comment]
> With [abdd675](https://github.com/s-u/OpenCL/commit/abdd6751aa973783bb5d4587b2dcfc35c5e0e491) we have more diagnostics, could you build that (e.g. master) and try again? Perhaps fetching the build log fails, we have so far just ignored that.
> 
> Other than that, have you tried running OpenCL workloads outside of R?

- I try to rebuild it, and again I get the same thing, i.e., no error message.

- I have not tried OpenCL outside of R.

**Comment by aaronpuchert** [comment]
Might be good to know what kind of platform/devices you have, could you run this and paste the result here?
```
lapply(oclPlatforms(), oclInfo)
lapply(oclDevices(), oclInfo)
```
Then before you compile the kernel, print `ctx`.

**Comment by s-u** [comment]
I __think__ this should be addressed by #13 since presumably the log was lost, now it will be part of the error.
